src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -145,7 +151,13 @@ public static FormulaValue DateValue_UO(IRContext irContext, UntypedObjectValue[
             if (impl.Type == FormulaType.String)
             {
                 var s = impl.GetString();
-                if (DateTime.TryParseExact(s, "yyyy-MM-dd", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime res))
+
+                if (!IsValidDateTimeUO(s))
+                {
+                    return CommonErrors.InvalidDateTimeError(irContext);
+                }
+
+                if (DateTime.TryParse(s, CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime res))
```

Comment:
[minor] can be combined as `if (IsValidDateTimeUO(s) && DateTime.TryParse(...))` - this way there will be a single return for `CommonErrors.InvalidDateTimeError(irContext)`

Reply:
Done